### ISSUE TITLE
For repreduction of baselines.

### ISSUE THREAD

**Issue by SwaggyZhang** [YuxiaWu/SimpleDyG#8]
Thank you for your previous help.

I am interested in your approach to adjust baselines, making them follow the evaluation paradigm of SimpleDyG.

Could you please share an example?

**Comment by YuxiaWu** [comment]
@SwaggyZhang I'm sorry for the late reply. The basic idea for baselines is that we compute the BPR loss after obtaining the embedding of the center nodes and positive/negative nodes from the backbone model. During evaluation, we rank all the candidate nodes to obtain top _K_ results.


Here is an example of the baselines from DyGLib (https://github.com/yule-BUAA/DyGLib). We refine the loss as follows:
```python
# The node embeddings are obtained from model[0] (the backbone model, such as JODIE, TGAT, etc)
# get positive and negative probabilities, shape (batch_size, )
positive_probabilities = model[1](input_1=batch_src_node_embeddings, input_2=batch_dst_node_embeddings).squeeze(dim=-1)#.sigmoid() # shape (batch_size, )
negative_probabilities = model[1](input_1=batch_neg_src_node_embeddings, input_2=batch_neg_dst_node_embeddings).squeeze(dim=-1)#.sigmoid() # shape (batch_size*neg_size, )

# repeat positive probabilities to match the shape of negative probabilities
positive_probabilities = positive_probabilities.repeat_interleave(repeats=args.neg_size, dim=0) # shape (batch_size*neg_size, )

predicts = torch.cat([positive_probabilities, negative_probabilities], dim=0)
labels = torch.cat([torch.ones_like(positive_probabilities), torch.zeros_like(negative_probabilities)], dim=0)
loss = -(positive_probabilities-negative_probabilities).sigmoid().log().sum()
```

**Comment by SwaggyZhang** [comment]
Thanks for sharing the code with me!
